src/datasource.ts

```diff
@@ -74,7 +74,7 @@ export class DataSource extends DataSourceWithBackend<Query, HistorianDataSource
 
         switch (target.queryType) {
           case 'AssetMeasurementQuery': {
-            const assetMeasurementQuery = { ...target.query } as AssetMeasurementQuery
+            const assetMeasurementQuery = JSON.parse(JSON.stringify(target.query)) as AssetMeasurementQuery
```

Comment:
? explain

Reply:
take a guess

Reply:
was to fix the annotations query bug that i changed it { ...target.query } to make sure changes to the query here are not visible/updated in the editor. JSON.parse and stringify because nested objects were still a problem

Reply:
aaaaaaah